Filters/MetadataOptionsFilter.cs

```diff
@@ -0,0 +1,34 @@
+using AngleSharp.Css;
+using Etch.OrchardCore.Greenhouse.Models;
+using Etch.OrchardCore.Greenhouse.Utilities;
+using Fluid;
+using Fluid.Values;
+using OrchardCore.Liquid;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Etch.OrchardCore.Greenhouse.Filters
+{
+    public class MetadataOptionsFilter : ILiquidFilter
+    {
+        public async ValueTask<FluidValue> ProcessAsync(FluidValue input, FilterArguments arguments, LiquidTemplateContext context)
+        {
+            var metadataField = new List<string>();
+            var propertyName = ((StringValue)arguments["property"]).ToObjectValue().ToString();
```

Comment:
Interesting, line 20 & 21 are both getting a string from the arguments, however they are doing it in two different ways. `arguments["selectedItem"].ToString()` is the cleanest so I'd suggest applying that approach to line 20.

Reply:
When changing line 21 to replicate 20, the value being passed is nested deeper in the object, when following line 20 approach i get back `Fluid.Values.StringValue`